### PR TITLE
Added TestProtocol in example_test

### PR DESCRIPTION
**Pull request by jzila**
Added TestProtocol in example_test to verify the example scenario. Also made a few stylistic fixes.

Signed-off-by: John Zila <john@jzila.com>

### CODE REVIEW

File: rpc2/dispatch.go [hunk 1]
```diff
@@ -12,16 +12,6 @@ type Dispatcher interface {
...
-type ResultPair struct {
-	res interface{}
-	err error
-}
-
-type ClientResultPair struct {
-	nxt DecodeNext
-	err error
-}
-
```
Comment (jzila) [inline rpc2/dispatch.go]: Doesn't seem to be used anywhere

File: rpc2/example/client.go [hunk 2]
```diff
@@ -14,32 +14,33 @@ type GenericClient interface {
 //---------------------------------------------------------------------
 
 type ArithClient struct {
-	cli GenericClient
+	GenericClient
```
Comment (jzila) [inline rpc2/example/client.go]: Just a stylistic preference: An `ArithClient` is a `GenericClient`, so it should adhere to the API.

File: rpc2/example/client.go [hunk 3]
```diff
@@ -14,32 +14,33 @@ type GenericClient interface {
...
 //---------------------------------------------------------------------
 
 type Client struct {
+	port int
 }
 
 func (s *Client) Run() (err error) {
 	var c net.Conn
-	if c, err = net.Dial("tcp", fmt.Sprintf("127.0.0.1:%d", port)); err != nil {
+	if c, err = net.Dial("tcp", fmt.Sprintf("127.0.0.1:%d", s.port)); err != nil {
```
Comment (jzila) [inline rpc2/example/client.go]: I'd prefer not to use package variables across file boundaries: it makes understanding the code and factoring out functionality more difficult.

File: rpc2/transport.go [hunk 4]
```diff
@@ -23,10 +23,10 @@ type Transporter interface {
 }
 
 type ConPackage struct {
+	Decoder
```
Comment (jzila) [inline rpc2/transport.go]: As above: a `ConPackage` is a `Decoder`, so it can meet the API.

**Comment by maxtaco** [comment]
Merged, thanks John!

**Pull request merged**